adapters/repos/db/vector/hnsw/search.go

```diff
@@ -83,6 +83,61 @@ func (h *hnsw) SearchByVector(ctx context.Context, vector []float32,
 	return h.knnSearchByVector(ctx, vector, k, h.searchTimeEF(k), allowList)
 }
 
+func (h *hnsw) SearchByMultiVector(ctx context.Context, vectors [][]float32, k int, allowList helpers.AllowList) ([]uint64, []float32, error) {
+	if !h.multivector.Load() {
+		return nil, nil, errors.New("multivector search is not enabled")
+	}
+
+	h.compressActionLock.RLock()
+	defer h.compressActionLock.RUnlock()
+
+	vectors = h.normalizeVecs(vectors)
+	flatSearchCutoff := int(atomic.LoadInt64(&h.flatSearchCutoff))
+	if allowList != nil && !h.forbidFlat && allowList.Len() < flatSearchCutoff {
+		return nil, nil, errors.New("flat search is not supported for multivector search")
```

Comment:
This flat search is just brute force if the filter is small. I don't think we can throw an error here and need to implement `flatSearch` function for multi vectors.